TL721X: Upgrade BLE SDK to Support A2

liucoldstar opened the pull request:
telink: tl721x: update hal_telink

andriy-bilynskyy commented:
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/cc7398accd3f47efef2e2116d3c48ec1acff0eb9 | https://github.com/telink-semi/hal_telink/commit/9576bc034dde35b43b8fd9f2f1e0bd89255f48f2 (develop) | [telink-semi/hal_telink@cc7398ac..9576bc03](https://github.com/telink-semi/hal_telink/compare/cc7398accd3f47efef2e2116d3c48ec1acff0eb9..9576bc034dde35b43b8fd9f2f1e0bd89255f48f2) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

andy-liu-telink committed:
telink: tl721x: update hal_telink

- update hal_telink revision
- make necessary changes
- update adc apis for tl721x

Signed-off-by: Andy Liu <jie.liu@telink-semi.com>

liucoldstar commented:
@s07641069 @andriy-bilynskyy 
Test results seem good for both Zephyr and Matter applications.
Could you take a look at this PR?

s07641069 reviewed (changes requested):
@liucoldstar @andy-liu-telink @damien0x0023 
Please create same PR into [develop_new](https://github.com/telink-semi/zephyr/tree/develop_new) branch and provide link in comment under this PR
In this case we can be sure that develop_new (Zephyr 3.7) build is not broken and that all changes are synchronized.

damien0x0023 commented:
> @liucoldstar @andy-liu-telink @damien0x0023 Please create same PR into [develop_new](https://github.com/telink-semi/zephyr/tree/develop_new) branch and provide link in comment under this PR In this case we can be sure that develop_new (Zephyr 3.7) build is not broken and that all changes are synchronized.

Add sync PR #442

andy-liu-telink reviewed (approved):
Approve
